drivers/SmartThings/zwave-garage-door-opener/src/ecolink-zw-gdo/init.lua

```diff
@@ -0,0 +1,458 @@
+    device.thread:call_with_delay(constants.DEFAULT_GET_STATUS_DELAY, function(d)
+      device:send(BarrierOperator:Get({}))
+    end)
+
+  else
+    log.info_with({hub_logs=true}, "<<send_open_to_gdo>> FAILED - SENSOR IN TAMPER")
+
+    device:emit_event_for_endpoint(GDO_ENDPOINT_NUMBER,
+                                  capabilities.doorControl.door.unknown()
+                                  )
+  end
+end
+
+local function send_close_to_gdo(driver, device, command)
+  if (
+    device:get_latest_state(CONTACTSENSOR_ENDPOINT_NAME,
+                            capabilities.tamperAlert.ID,
+                            capabilities.tamperAlert.tamper.NAME) == "clear"
+  ) then
+    log.info_with({hub_logs=true}, "<<send_close_to_gdo>> - SUCCESS")
+
+    device:send(BarrierOperator:Set({ target_value = BarrierOperator.state.CLOSED }))
+    device.thread:call_with_delay(constants.DEFAULT_GET_STATUS_DELAY, function(d)
+      device:send(BarrierOperator:Get({}))
+    end)
+else
+  log.info_with({hub_logs=true}, "<<send_close_to_gdo>> FAILED - SENSOR IN TAMPER")
+
+  device:emit_event_for_endpoint(GDO_ENDPOINT_NUMBER,
+                                capabilities.doorControl.door.unknown()
+                                )
+end
+end
+
+
+--- Handle Device Instantiated Event
+---
+--- @param driver st.zwave.Driver
+--- @param device st.zwave.Device
+local function device_instatiated(driver, device)
+  log.info_with({hub_logs=true}, "device init")
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+  device:set_endpoint_to_component_fn(endpoint_to_component)
+  -- device:subscribe()
+
+  device.thread:call_with_delay(constants.DEFAULT_GET_STATUS_DELAY, function(d)
+    device:send(BarrierOperator:Get({}))
+  end)
+  device.thread:call_with_delay(constants.DEFAULT_GET_STATUS_DELAY*2, function(d)
+    device:send(SensorMultilevel:Get({}))
+  end)  
+  device.thread:call_with_delay(constants.DEFAULT_GET_STATUS_DELAY*3, function(d)
+    device:send(Configuration:BulkGetV2({parameter_offset = 1, number_of_parameters = 6}) )
+  end)
+
+  -- Reset contact sensor battery level... This should be pollable on the GDO side.
+  device:emit_event_for_endpoint(CONTACTSENSOR_ENDPOINT_NUMBER,
+                                capabilities.battery.battery(CONTACTSENSOR_BATTERY_LEVEL_NORMAL)
+                                )
+  -- Reset contact sensor fields
+  device:emit_event_for_endpoint(CONTACTSENSOR_ENDPOINT_NUMBER,
+                                capabilities.tamperAlert.tamper.clear()
+                                )
+  device:emit_event_for_endpoint(CONTACTSENSOR_ENDPOINT_NUMBER,
+                                capabilities.contactSensor.contact.closed()
+                                )
+
+  -- Init barrier door state
+  device:emit_event_for_endpoint(GDO_ENDPOINT_NUMBER,
+                                capabilities.doorControl.door.closed()
+                                )
+
```

Comment:
the `init` event happens every time the driver starts up, so for example on a hub restart. Are you sure you want to be emitting these synthetic events in this circumstance? typically we do things like this in `added` so they only happen once

Reply:
I thought it worked in the opposite way. Thanks for clearing that up. Moved to `device_added`